spackbot/handlers/style.py

```diff
@@ -23,3 +31,99 @@ async def style_comment(event, gh):
                 number,
             )
             await gh.post(comments_url, {}, data={"body": comments.style_message})
+
+
+def is_up_to_date(output):
+    """
+    A commit can fail if there are no changes!
+    """
+    return "branch is up to date" in output
+
+
+async def fix_style(event, gh):
+    """
+    Respond to a request to fix style.
+    We first retrieve metadata about the pull request. If the request comes
+    from anyone with write access to the repository, we commit, and we commit
+    under the identity of the original person that opened the PR.
+    """
+    response = requests.get(event.data["issue"]["pull_request"]["url"])
```

Comment:
I think we already discussed this is out of scope and can be tried with another PR - the requests is uesd for a reason (I don't think the async call with the authenticated client worked here) so I defaulted back to simple. KISS principle!

Reply:
I’d like you to try to figure this one out. Importing another library to do something aiohttp should be able to do is not KISS IMO. We should figure out the deeper problem instead of doing a workaround like this. Not seeing why the authenticated request would fail because of authentication.

Reply:
yes but requests is used throughout the library in other handlers not relevant to this PR. If it needs to be fixed that's okay but it's out of scope for here.

Reply:
> Not seeing why the authenticated request would fail because of authentication.

There are many cases in the wild where adding authentication actually makes it not work. Not sure if that is the issue here, but that's definitely a thing.